crates/zkvm-jetpack/src/jets/verifier_jets.rs

```diff
@@ -176,3 +202,202 @@ fn process_belt(
+    // |=  $:  field=?(%ext %base)
+    //         mp=mp-mega
+    //         args=bpoly  :: can be bpoly or fpoly
+    //         chal-map=(map @ belt)
+    //         dyns=bpoly
+    //         com-map=(map @ elt)
+    //     ==
+    // ^-  elt
+    let sam = slot(subject, 6)?;
+    let stack = &mut context.stack;
+    let [field, mp, args, chal_map, dyns, com_map] = sam.uncell()?;
+
+    let Ok(dyns) = BPolySlice::try_from(dyns) else {
+        return jet_err();
+    };
+
+    let ret = match field.as_direct()?.data() {
+        tas!(b"ext") => mpeval::<Felt>(stack, mp, args, chal_map, dyns, com_map)?.to_noun(stack),
+        tas!(b"base") => mpeval::<Belt>(stack, mp, args, chal_map, dyns, com_map)?.to_noun(stack),
+        _ => return jet_err(),
+    };
+
+    Ok(ret)
+}
+
+fn mpeval<F: Fops>(
+    stack: &mut NockStack,
+    mp: Noun,
+    args: Noun,
+    chal_map: Noun,
+    dyns: BPolySlice,
+    com_map: Noun,
+) -> Result<F, JetErr>
+where
+    for<'a> PolySlice<'a, F>: TryFrom<Noun>,
+{
+    let Ok(args) = PolySlice::try_from(args) else {
+        return jet_err();
+    };
+
+    let chal_map = HoonMap::try_from(chal_map).ok();
+    let com_map = HoonMap::try_from(com_map).ok();
+
+    // ?:  =(~ mp)
+    if mp.is_atom() {
+        // init-zero
+        return Ok(F::zero());
+    }
+
+    // %+  roll  ~(tap by mp)
+    // |=  [[k=bpoly v=belt] acc=_init-zero]
+    let mut mp = HoonMapIter::from(mp);
+
+    mp.try_fold(F::zero(), |acc, n| {
+        let [k, v] = n.uncell()?;
+
+        let Ok(k) = BPolySlice::try_from(k) else {
+            return jet_err();
+        };
+        let v = Belt::from_noun(v)?;
+        // =/  coeff=@ux  (lift-op v)
+        let coeff = F::lift(v);
+        // ?:  =(init-zero coeff)
+        if coeff == F::zero() {
+            // acc
+            return Ok(acc);
+        }
+
+        // %+  add-op  acc
+        // %+  mul-op  coeff
+        // %+  roll  (range len.k)
+        // |=  [i=@ res=_init-one]
+        // ?:  =(init-zero res)
+        //   init-zero
+        let res = k
+            // =/  ter  (~(snag bop k) i)
+            .iter()
+            .copied()
+            // =/  [typ=mega-typ idx=@ exp=@ud]  (brek ter)
+            .map(brek)
+            .map(|(typ, idx, exp)| {
+                // ?-  typ
+                match typ {
+                    //     %var
+                    MegaTyp::Var => {
+                        //   %+  pow-op
+                        //     (~(snag aop-door args) idx)
+                        //   exp
+                        args.0[idx].pow(exp)
+                    }
+                    // ::
+                    //     %rnd
+                    MegaTyp::Rnd => {
+                        //   %+  pow-op
+                        //     (lift-op (~(got by chal-map) idx))
+                        //   exp
+                        let v = chal_map
+                            .as_ref()
+                            .unwrap()
```

Comment:
Our style is to never use unwrap(). Either use ? to return an error up, or use expect() with a message.